addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -1348,26 +1348,23 @@ export class OdooEditor extends EventTarget {
         }
     }
 
-    setContenteditableLink(link) {
-        const editableChildren = link.querySelectorAll('[contenteditable=true]');
-        this._stopContenteditable();
-
-        this._fixLinkMutatedElements = {
-            link,
-            wasContenteditableTrue: [...editableChildren],
-            wasContenteditableFalse: [],
-            wasContenteditableNull: [],
-        };
-        const contentEditableAttribute = link.getAttribute('contenteditable');
-        if (contentEditableAttribute === 'true') {
-            this._fixLinkMutatedElements.wasContenteditableTrue.push(link);
-        } else if (contentEditableAttribute === 'false') {
-            this._fixLinkMutatedElements.wasContenteditableFalse.push(link);
-        } else {
-            this._fixLinkMutatedElements.wasContenteditableNull.push(link);
+    _setLinkZws() {
+        this._resetLinkZws();
+        const linkInSelection = getInSelection(this.document, 'a');
+        for (const link of this.editable.querySelectorAll('a')) {
+            if (!isBlock(link) && link.textContent !== '' && !(link.children.length === 1 && link.firstElementChild.nodeName === 'IMG')) {
```

Comment:
I'm ok either way but I do still prefer the `!(A && B)`, because it's semantically more straight forward to me. I want to say "ignore links wrapped around a single image", or "ignore links with a single element child which is an image", or "do this only for links that don't have a single element child which is an image", not "do this only for link with any number of element children but one, unless that single element child is an image". The latter feels like I'm separating the statements while I'm trying to say only one thing, see what I mean? I'm _excluding a group_, not _including two statements_.
There definitely should be a comment though.

Reply:
@Zinston Let's go for something like the last proposition in my original message then :)

Reply:
@dmo-odoo Yes I agree that's the best one :-)